bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/osgi/ResourceBundleClassLoader.java

```diff
@@ -36,10 +36,15 @@
  * mappings.
  *
  * @author Michael Grammling - Initial Contribution
+ * @author Martin Herbst - ISO-8859-1 support added
  */
 public class ResourceBundleClassLoader extends ClassLoader {
 
-    private static final String[] SUPPORTED_CHARSETS = { "UTF-8" };
+    /**
+     * ISO-8859-1 must be the last array element because all files could be decoded to ISO-8859-1 and
+     * therefore always character set ISO-8859-1 would be assumed.
```

Comment:
> but only the character detection method detectCharset.

Yes, that was my idea.
It could be made visible within the package (i.e. default visibility), then the tests could access it without any reflection tricks.